tee-worker/scripts/litentry/release/deploy.sh

```diff
@@ -1,736 +1,555 @@
-        target="${target_dir}/${file}"
-
-        if [ -f "$source" ]; then
-            cp "$source" "$target"
-        else
-            echo "$source does not exist, this is fine, but you can't perform remote attestation with this."
-        fi
-    done
-
-    for Item in 'enclave.signed.so' 'litentry-worker' 'aes_key_sealed.bin' 'ed25519_key_sealed.bin' 'enclave-shielding-pubkey.json' 'enclave-signing-pubkey.bin' 'rsa3072_key_sealed.bin' 'sidechain_db'; do
-      cp -r "${Item}" "${target_dir}"
-    done
-
-    # Only possible in TEE-Internal
-    cp $CONFIG "${target_dir}/mode_config.json"
-    if [ "$PRODUCTION" = true ]; then
-      cp $INTEL_KEY "${target_dir}/key_production.txt"
-      cp $INTEL_SPID "${target_dir}/spid_production.txt"
-    else
-      cp $INTEL_KEY "${target_dir}/key.txt"
-      cp $INTEL_SPID "${target_dir}/spid.txt"
-    fi
-}
-
-function restart_worker() {
-  # Need to make sure we have the JSON
-  cd $ROOTDIR || exit
-  worker_count=$(echo "$CONFIG" | jq '.workers | length')
-
-  for ((i = 0; i < worker_count; i++)); do
-    WORKER_DIR=/opt/worker/w${i}
-    # Remove previous logs if any
-    rm -r $ROOTDIR/tee-worker/log/worker${i}.log
-    # Prepare the Worker Directory before restarting
-    mkdir -p /opt/worker/w${i}
-    setup_working_dir $ROOTDIR/tee-worker/bin /opt/worker/w${i}
-
-    # We only need this in productive enclave
-    if [ "$PRODUCTION" = true ]; then
-      # Transfer balance to the enclave account that is generated
-      echo "Transferring balance to the enclave account"
-      cd $ROOTDIR/scripts/ts-utils/ || exit
-      yarn install
-      npx ts-node transfer.ts  $ENCLAVE_ACCOUNT
+    if [ "$PRODUCTION" = false ]; then
+      for f in 'key.txt' 'spid.txt'; do
+        [ -f "$f" ] || touch "$f"
+      done
     fi
 
-    cd $ROOTDIR/tee-worker || exit
-
-    source=$(echo "$CONFIG" | jq -r ".workers[$i].source")
-    flags=$(echo "$CONFIG" | jq -r ".workers[$i].flags[]")
-    subcommand_flags=$(echo "$CONFIG" | jq -r ".workers[$i].subcommand_flags[]")
+    for ((i = 0; i < WORKER_COUNT; i++)); do
+      worker_dir="$WORKER_BASEDIR/w$i"
+      mkdir -p "$worker_dir"
+      for f in 'key.txt' 'spid.txt' 'enclave.signed.so' 'litentry-worker'; do
+        [ -f "$f" ] && cp -f "$f" "$worker_dir"
+      done
 
-    command="./litentry-worker"
+      cd "$worker_dir"
+      [ -f light_client_db.bin/db.bin.backup ] && cp -f light_client_db.bin/db.bin.backup light_client_db.bin/db.bin
 
-    for flag in $flags; do
-      command+=" $flag"
-    done
+      enclave_account=$(./litentry-worker signing-key | grep -oP '^Enclave account: \K.*$$')
 
-    command+=" run"
-
-    for subcommand_flag in $subcommand_flags; do
-      command+=" $subcommand_flag"
+      if [ "$PRODUCTION" = true ]; then
+        echo "Transferring balance to the enclave account $enclave_account ..."
+        cd $ROOTDIR/scripts/ts-utils/ || exit
+        yarn install
+        npx ts-node transfer.ts $enclave_account
```

Comment:
If we build the parachain without the `tee-dev` feature, We will also have to schedule the enclave from the first block itself, else the `register_enclave` call by the `worker` will fail.

Reply:
I think this script can't be used for a full automated prod development anyway as we don't know the teerex admin, as written in the PR description ...

Reply:
I'm fine with that - actually, the prod parachain should not be reset once in a while (once we move to at least rococo), so actually we only have to take care of the worker restart